pubsub/cloud-client/subscriber.py

```diff
@@ -72,16 +72,65 @@ def create_subscription(project_id, topic_name, subscription_name):
         project_id, subscription_name
     )
 
-    subscription = subscriber.create_subscription(
-        subscription_path, topic_path
-    )
+    subscription = subscriber.create_subscription(subscription_path, topic_path)
 
     print("Subscription created: {}".format(subscription))
 
     subscriber.close()
     # [END pubsub_create_pull_subscription]
 
 
+def create_subscription_with_dead_letter_topic(
+    project_id, topic_name, subscription_name, dead_letter_topic_name
+):
+    """Create a subscription with dead letter policy."""
+    # [START pubsub_dead_letter_create_subscription]
+    from google.cloud import pubsub_v1
+    from google.cloud.pubsub_v1.types import DeadLetterPolicy
+
+    # TODO project_id = "Your Google Cloud Project ID"
+    # This is an existing topic that the subscription with dead letter policy
+    # is attached to.
+    # TODO topic_name = "Your Pub/Sub topic name"
+    # This is an existing subscription with a dead letter policy.
+    # TODO subscription_name = "Your Pub/Sub subscription name"
+    # This is an existing dead letter topic that the subscription with dead
+    # letter policy will forward dead letter messages to.
+    # TODO dead_letter_topic_name = "Your Pub/Sub dead letter topic name"
+
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project_id, topic_name)
+    subscription_path = subscriber.subscription_path(
+        project_id, subscription_name
+    )
+    dead_letter_topic_path = subscriber.topic_path(
+        project_id, dead_letter_topic_name
+    )
+
+    dead_letter_policy = DeadLetterPolicy(
+        dead_letter_topic=dead_letter_topic_path, max_delivery_attempts=10
+    )
+
+    subscription = subscriber.create_subscription(
+        subscription_path, topic_path, dead_letter_policy=dead_letter_policy
+    )
+
+    print("Subscription created: {}".format(subscription.name))
+    print(
+        "It will forward dead letter messages to: {}".format(
+            subscription.dead_letter_policy.dead_letter_topic
+        )
+    )
+    print(
+        "After {} delivery attempts.".format(
+            subscription.dead_letter_policy.max_delivery_attempts
+        )
+    )
+
+    subscriber.close()
```

Comment:
What does closing this subscriber do? Should we use a `with subscriber:` statement to ensure it's closed if an error happens? 

It this subscriber being used? It's not clear to me what this sample is doing.

Reply:
Done.